### ISSUE TITLE
Fail to load custom plugins: plugin was built with a different version of package

### ISSUE THREAD

**Issue by self-cream** [volcano-sh/volcano#3174]
**What happened**:
I am trying to implement a custom plugin following the instruction of custom-plugin.md in the docs. However, the built pluginable volcano scheduler raised an error:
```
2023/11/04 17:37:44 maxprocs: Leaving GOMAXPROCS=48: CPU quota undefined
W1104 17:37:44.641031       1 client_config.go:617] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
E1104 17:37:44.690243       1 server.go:72] Fail to load custom plugins: plugin.Open("plugins/horus"): plugin was built with a different version of package volcano.sh/volcano/pkg/scheduler/util/assert
plugin.Open("plugins/horus"): plugin was built with a different version of package volcano.sh/volcano/pkg/scheduler/util/assert
```
I searched a lot for a solution to this problem. Most of them considered that this error is caused by the inconsistent version of sharing library between the scheduler and the plugin. However, this package (volcano.sh/volcano) is only recorded in the go.mod of my plugin project. By the way, I didn't use vendor to manage the dependencies of the scheduler and the plugin. I am wondering what is the cause of this error.

**What you expected to happen**:
I hope anyone could give me some advices or possible solutions.

**How to reproduce it (as minimally and precisely as possible)**:
1. I used ubuntu as the base docker image, and I followed the instructions in custom-plugin.md to build a pluginable scheduler.
```
make images SUPPORT_PLUGINS=yes
```
2. I implemented my plugin and used ubuntu as the base docker image to build my plugin. The building command is shown as follows:
```
CGO_ENABLED=1 go build -mod=readonly -o plugins/horus.so -buildmode=plugin horus.go
```
3. I wrote a Dockerfile to add my plugin to the built pluginable volcano scheduler. Here is the Dockerfile:
```
FROM volcanosh/vc-scheduler:v1.6.0

COPY plugins plugins
```

4. I made some modifications to the volcano scheduler Deployment in volcano-deployment.yaml. These modifications are shown as follows:
```
containers:
        - name: volcano-scheduler
          image: registry.cn-hangzhou.aliyuncs.com/njupt-lzj/vc-scheduler:v1.6.0
          command:
            - /vc-scheduler
          args:
            - --logtostderr
            - --scheduler-conf=/volcano.scheduler/volcano-scheduler.conf
            - --enable-healthz=true
            - --enable-metrics=true
            - -v=3
            - --plugins-dir=plugins
            - 2>&1
```
5. I viewed the log of the volcano-scheduler, and it raised an error:
```
2023/11/04 17:37:44 maxprocs: Leaving GOMAXPROCS=48: CPU quota undefined
W1104 17:37:44.641031       1 client_config.go:617] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
E1104 17:37:44.690243       1 server.go:72] Fail to load custom plugins: plugin.Open("plugins/horus"): plugin was built with a different version of package volcano.sh/volcano/pkg/scheduler/util/assert
plugin.Open("plugins/horus"): plugin was built with a different version of package volcano.sh/volcano/pkg/scheduler/util/assert
```
**Anything else we need to know?**:
I didn't use vendor to manage the dependencies of the scheduler and the plugin. According to my search results, this error (plugin was built with a different version of package xxx) only occurs when people use vendor to manage the dependencies. Meanwhile, the package (volcano.sh/volcano) is only recorded in the go.mod of my plugin project, which is not the sharing library between the scheduler and the plugin. So, I am wondering what is the cause of this error.
**Environment**:
- Volcano Version: v1.6.0
- Kubernetes version (use `kubectl version`): v1.23.0
- Cloud provider or hardware configuration: Intel(R) Xeon(R) Silver 4214 CPU @ 2.20GHz, 64GB RAM
- OS (e.g. from /etc/os-release): Centos 7
- Kernel (e.g. `uname -a`): Linux 3.10.0-1160.49.1.el7.x86_64
- Install tools: go 1.18.10, docker-ce 19.03
- Others:

**Comment by self-cream** [comment]
This problem has been solved. The cause is that the path of this package (volcano.sh/volcano) should be consistent with the directory where volcano-scheduler built. So, what I did is update the go.mod file of my plugin project:
```
replace (
              volcano.sh/volcano => /volcano # your local path of volcano directory
)
```

**Comment by murali1539** [comment]
@self-cream I am facing the same issue with my custom plugin development. I tried the solution you have provided, but it did not work for me. I am thinking it could be due to project setup. Did you place the "volcano" code inside your plugin code's folder ? I have placed the volcano source code in a different folder and facing the following error : 

`plugin was built with a different version of package k8s.io/kube-openapi/pkg/internal/third_party/go-json-experiment/json`

But the version of `k8s.io/kube-openapi` is exactly same in plugin project as well as my plugin repo.

Another question : 
In step 1, you have built a docker image for vc-scheduler. So, it should be a different image then the official image, right ? Because in Step 3, you are using the official docker image of volcano, where the SUPPORT_PLUGINS=yes is not set. How is this working ?
